pages/index.js

```diff
@@ -6,11 +9,18 @@ import { useEffect, useState, useRef } from "react";
 import { useRouter } from "next/router";
 import styled from "styled-components";
 
-export async function getServerSideProps() {
-  const days = await getAllDays();
-  return {
-    props: { days: JSON.parse(JSON.stringify(days)) },
-  };
+export async function getServerSideProps(context) {
+  const session = await unstable_getServerSession(
+    context.req,
+    context.res,
+    authOptions
+  );
+  if (session) {
+    const days = await getAllDays(session.user.email);
+    return {
+      props: { days: JSON.parse(JSON.stringify(days)) },
+    };
+  } else return { props: {} };
```

Comment:
did it not make days undefined?

Reply:
only if there is no session (so when nobody is logged in) then it also doesn't render the stuff, which needs the props, thus thats how it's supposed to be